lib/remote/jsonrpcconnection-pki.cpp

```diff
@@ -72,8 +72,10 @@ Value RequestCertificateHandler(const MessageOrigin::Ptr& origin, const Dictiona
 		 * serial number. */
 		time_t forceRenewalEnd = 1483228800; /* January 1st, 2017 */
 		time_t renewalStart = now + 30 * 24 * 60 * 60;
+		ASN1_TIME *crtExpired = X509_get_notBefore(cert.get()) + 397 * 24 * 60 * 60;
 
-		if (X509_cmp_time(X509_get_notBefore(cert.get()), &forceRenewalEnd) != -1 && X509_cmp_time(X509_get_notAfter(cert.get()), &renewalStart) != -1) {
+		if ((X509_cmp_time(X509_get_notBefore(cert.get()), &forceRenewalEnd) != -1 && X509_cmp_time(X509_get_notAfter(cert.get()), &renewalStart) != -1)
+			|| X509_get_notAfter(cert.get()) >= crtExpired) {
```

Comment:
What's `crtExpired`'s datatype?

Reply:
This is openssl standard date type for timestamp like `std::time_t`.

Reply:
Yes, but it's _a pointer_. Are you sure about what exactly do you do with that pointer?